src/sonic-yang-models/yang-models/sonic-pbh.yang

```diff
@@ -0,0 +1,250 @@
+	}
+
+	typedef hash-field {
+		description "Represents native hash field";
+		type enumeration {
+			enum INNER_IP_PROTOCOL;
+			enum INNER_L4_DST_PORT;
+			enum INNER_L4_SRC_PORT;
+			enum INNER_DST_IPV4;
+			enum INNER_SRC_IPV4;
+			enum INNER_DST_IPV6;
+			enum INNER_SRC_IPV6;
+		}
+	}
+
+	typedef packet-action {
+		description "Represents packet action";
+		type enumeration {
+			enum SET_ECMP_HASH;
+			enum SET_LAG_HASH;
+		}
+	}
+
+	typedef flow-counter {
+		description "Represents flow counter state";
+		type enumeration {
+			enum DISABLED;
+			enum ENABLED;
+		}
+	}
+
+	container sonic-pbh {
+
+		container PBH_HASH_FIELD {
+
+			description "PBH_HASH_FIELD part of config_db.json";
+
+			list PBH_HASH_FIELD_LIST {
+
+				key "hash_field_name";
+
+				leaf hash_field_name {
+					description "The name of this hash field";
+					type string;
+				}
+
+				leaf hash_field {
+					description "Configures native hash field for this hash field";
+					mandatory true;
+					type pbh:hash-field;
+				}
+
+				leaf ip_mask {
+					description "Configures IPv4/IPv6 address mask for this hash field";
+					mandatory true;
+					type union {
+						type inet:ipv4-address;
+						type inet:ipv6-address;
+					}
+					when "boolean(../hash_field[.='INNER_DST_IPV4' or .='INNER_SRC_IPV4' or .='INNER_DST_IPV6' or .='INNER_SRC_IPV6'])";
+				}
+
+				leaf sequence_id {
+					description "Configures in which order the fields are hashed and defines which fields should be associative";
+					mandatory true;
+					type uint32;
+				}
+
+			}
+			/* end of PBH_HASH_FIELD_LIST */
+		}
+		/* end of container PBH_HASH_FIELD */
+
+		container PBH_HASH {
+
+			description "PBH_HASH part of config_db.json";
+
+			list PBH_HASH_LIST {
+
+				key "hash_name";
+
+				leaf hash_name {
+					description "The name of this hash";
+					type string;
+				}
+
+				leaf-list hash_field_list {
+					description "The list of hash fields to apply with this hash";
+					min-elements 1;
+					type leafref {
+						path "/pbh:sonic-pbh/pbh:PBH_HASH_FIELD/pbh:PBH_HASH_FIELD_LIST/pbh:hash_field_name";
+					}
+				}
+
+			}
+			/* end of PBH_HASH_LIST */
+		}
+		/* end of container PBH_HASH */
+
+		container PBH_RULE {
+
+			description "PBH_RULE part of config_db.json";
+
+			list PBH_RULE_LIST {
+
+				key "table_name rule_name";
+
+				leaf table_name {
+					description "The name of table which holds this rule";
+					type leafref {
+						path "/pbh:sonic-pbh/pbh:PBH_TABLE/pbh:PBH_TABLE_LIST/pbh:table_name";
+					}
+				}
+
+				leaf rule_name {
+					description "The name of this rule";
+					type string;
+				}
+
+				leaf priority {
+					description "Configures priority for this rule";
+					mandatory true;
+					type uint32;
+				}
+
+				leaf gre_key {
+					description "Configures packet match for this rule: GRE key (value/mask)";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
+					}
+				}
+
+				leaf ip_protocol {
```

Comment:
We need to look at the possibility to remove the mask from next 4 fields....Please check with Anish. Thx.